src/test/java/com/epam/tc/hw9/ListsTest.java

```diff
@@ -0,0 +1,111 @@
+import com.example.types.BoardDto;
+import com.example.types.ListDto;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import java.util.Arrays;
+import org.assertj.core.api.SoftAssertions;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+public class ListsTest {
+
+    private BoardDto board;
+
+    private RestListsService listsService;
+    private RestBoardsService boardsService;
+
+    @BeforeMethod
+    public void init() {
+        boardsService = new RestBoardsService();
+        listsService = new RestListsService();
+        board = boardsService
+            .createBoard(DefaultObjectCreator.createBoard());
+    }
+
+    @AfterMethod
+    public void clean() {
+        Arrays.stream(boardsService.getBoards())
+              .map(BoardDto::getId)
+              .forEach(boardsService::deleteBoard);
+    }
+
+    @Test(description = "Create default list.")
+    public void createList() {
+        ListDto listForCreation = DefaultObjectCreator.createList();
+
+        String createdListId = listsService.createList(listForCreation.withIdBoard(board.getId()))
+                                           .getId();
+
+        ListDto createdList = listsService.getList(createdListId);
+
+        new RestListAssertions(createdList)
+            .verifyListName(listForCreation.getName())
+            .verifyListId(createdListId);
+    }
+
+    @Test(description = "Create lists on the specified boards.",
+          dataProviderClass = TrelloDataProvider.class,
+          dataProvider = "listNames")
+    public void createListsOnBoard(String testId, String boardName, String... lists) {
+        for (String listName : lists) {
+            board = boardsService.createBoard(DefaultObjectCreator.createBoard().withName(boardName));
+
+            ListDto listForCreation = DefaultObjectCreator.createList()
+                                                          .withName(listName)
+                                                          .withIdBoard(board.getId());
+
+            String createdListId = listsService.createList(listForCreation)
+                                               .getId();
+
+            ListDto createdList = listsService.getList(createdListId);
+
+            new RestListAssertions(createdList)
+                .verifyListName(listForCreation.getName())
+                .verifyListId(createdListId);
+        }
+    }
+
+    @Test
+    public void deleteBoardWithList() {
+        SoftAssertions softAssertions = new SoftAssertions();
+
+        String listId = listsService.createList(DefaultObjectCreator.createList().withIdBoard(board.getId()))
+                                    .getId();
+
+        String response = boardsService.deleteBoard(board.getId());
+        checkDeleted(response, softAssertions);
+
+        softAssertions.assertThatThrownBy(() -> listsService.getList(listId))
```

Comment:
Better to verify this case with GET method which expects to get ERROR_CODE instead of 200 OK